src/Domain/UseCase/SyncItems.ts

```diff
@@ -21,18 +23,38 @@ export class SyncItems implements UseCaseInterface {
       contentType: dto.contentType
     })
 
-    const saveItemsResult = await this.itemService.saveItems(
-      dto.itemHashes,
-      dto.userAgent,
-      getItemsResult.items
-    )
+    const saveItemsResult = await this.itemService.saveItems({
+      itemHashes: dto.itemHashes,
+      userAgent: dto.userAgent,
+      userUuid: dto.userUuid
+    })
+
+    let retrievedItems = this.filterOutSyncConflictsForConsecutiveSyncs(getItemsResult.items, saveItemsResult.conflicts)
+    if (this.isFirstSync(dto)) {
+      retrievedItems = await this.itemService.frontLoadKeysItemsToTop(dto.userUuid, retrievedItems)
+    }
 
     return {
-      retrievedItems: getItemsResult.items,
-      savedItems: saveItemsResult.items,
-      conflicts: saveItemsResult.conflicts,
+      retrievedItems,
       syncToken: saveItemsResult.syncToken,
+      savedItems: saveItemsResult.savedItems,
+      conflicts: saveItemsResult.conflicts,
       cursorToken: getItemsResult.cursorToken
     }
   }
+
+  private isFirstSync(dto: SyncItemsDTO): boolean {
+    return dto.syncToken === undefined
+  }
+
+  private filterOutSyncConflictsForConsecutiveSyncs(retrievedItems: Array<Item>, conflicts: Array<ItemConflict>): Array<Item> {
+    const syncConflictIds: Array<string> = []
+    conflicts.map((conflict: ItemConflict) => {
```

Comment:
Should use `forEach` maybe? I've only ever seen map used when you care about the return value.

Reply:
Will fix 👍